src/tokenizers.js

```diff
@@ -20,6 +20,8 @@
  * @module tokenizers
  */
 
+import XRegExp from 'xregexp';
```

Comment:
Is RegExp not supported in react native?

Reply:
RegExp yes, unicode no it seems... I wonder if this should just be handled like a polyfill and keep `new RegExp`... no one but React Native users need that extra lib I guess

Reply:
Yeah I don't want to include another dependency. Polyfill sounds best.

Reply:
But seems no polyfill package for unicode.
Maybe I create a new package.

Reply:
First I thought something like this would be possible:

```js
globalThis.RegExp = (await import("xregexp")).default;
```

But:

```js
> new RegExp(/123/)
Uncaught RangeError: Maximum call stack size exceeded
    at new XRegExp (/var/www/html/transformer/transformers.js_es6/reactnative/node_modules/xregexp/lib/xregexp.js:625:3)
    at new XRegExp (/var/www/html/transformer/transformers.js_es6/reactnative/node_modules/xregexp/lib/xregexp.js:705:18)
    at new XRegExp (/var/www/html/transformer/transformers.js_es6/reactnative/node_modules/xregexp/lib/xregexp.js:705:18)
    at new XRegExp (/var/www/html/transformer/transformers.js_es6/reactnative/node_modules/xregexp/lib/xregexp.js:705:18)
    at new XRegExp (/var/www/html/transformer/transformers.js_es6/reactnative/node_modules/xregexp/lib/xregexp.js:705:18)
    at new XRegExp (/var/www/html/transformer/transformers.js_es6/reactnative/node_modules/xregexp/lib/xregexp.js:705:18)
    at new XRegExp (/var/www/html/transformer/transformers.js_es6/reactnative/node_modules/xregexp/lib/xregexp.js:705:18)
    at new XRegExp (/var/www/html/transformer/transformers.js_es6/reactnative/node_modules/xregexp/lib/xregexp.js:705:18)
    at new XRegExp (/var/www/html/transformer/transformers.js_es6/reactnative/node_modules/xregexp/lib/xregexp.js:705:18)
    at new XRegExp (/var/www/html/transformer/transformers.js_es6/reactnative/node_modules/xregexp/lib/xregexp.js:705:18)
```

XRegExp depends itself on RegExp...

I think you could still do:

```sh
mkdir reactnative
cd reactnative
npm init
npm i xregexp
```

And then the dependency is limited to reactnative only... plus this polyfill that I just made up / tested:

```js
const XRegExp = (await import("xregexp")).default;
const originalRegExp = globalThis.RegExp;
function polyfillRegExp(...args) {
    // Fill in original RegExp, otherwise: Uncaught RangeError: Maximum call stack size exceeded
    globalThis.RegExp = originalRegExp;
    const ret = new XRegExp(...args);
    globalThis.RegExp = polyfillRegExp; // polyfill back...
    return ret;
}
globalThis.RegExp = polyfillRegExp;
```

Test:

![image](https://github.com/xenova/transformers.js/assets/5236548/cd2591c4-0c1e-4725-9068-58510f4b474d)